airsonic-main/src/main/java/org/airsonic/player/service/search/IndexManager.java

```diff
@@ -313,8 +313,10 @@ private void stopIndexing(IndexType type, MediaLibraryStatistics statistics) {
                 }
             } catch (IndexNotFoundException e) {
                 LOG.debug("Index {} does not exist in {}, likely not yet created.", indexType.toString(), indexDirectory.getAbsolutePath());
+                return null;
             } catch (IOException e) {
                 LOG.warn("Failed to initialize SearcherManager.", e);
+                return null;
```

Comment:
After the exception is caught, subsequent processing should not be performed.
As the logs show, it seems that references to already healthy Serchers are being dropped.